Why do I get this error:

```
/g/data1b/jh2/ps3459/quokka/src/EOS.hpp:118:23: error: ‘Tgas’ was not declared in this scope
  118 |         chemstate.T = Tgas;

```

--- a/src/EOS.hpp
+++ b/src/EOS.hpp
@@ -66,7 +66,7 @@ AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE auto EOS<problem_t>::ComputeTgasFromEin
 	chemstate.e = Eint / rho;
 
 	if (massScalars.has_value()) {
-		const auto& massArray = massScalars.value();
+		const auto &massArray = massScalars.value();
 		for (int nn = 0; nn < nmscalars_; ++nn) {
 			chemstate.xn[nn] = massArray[nn] / spmasses[nn]; // massScalars are partial densities (massFractions * rho)
 		}
@@ -96,7 +96,7 @@ AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE auto EOS<problem_t>::ComputeEintFromTga
 	chemstate.T = Tgas;
 
 	if (massScalars.has_value()) {
-		const auto& massArray = massScalars.value();
+		const auto &massArray = massScalars.value();
 		for (int nn = 0; nn < nmscalars_; ++nn) {
 			chemstate.xn[nn] = massArray[nn] / spmasses[nn]; // massScalars are partial densities (massFractions * rho)
 		}
@@ -126,7 +126,7 @@ AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE auto EOS<problem_t>::ComputeEintTempDer
 	chemstate.T = Tgas;
 
 	if (massScalars.has_value()) {
-		const auto& massArray = massScalars.value();
+		const auto &massArray = massScalars.value();
 		for (int nn = 0; nn < nmscalars_; ++nn) {
 			chemstate.xn[nn] = massArray[nn] / spmasses[nn]; // massScalars are partial densities (massFractions * rho)
 		}

--- a/src/EOS.hpp
+++ b/src/EOS.hpp
@@ -93,7 +93,9 @@ AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE auto EOS<problem_t>::ComputeEintFromTga
 #ifdef PRIMORDIAL_CHEM
 	burn_t chemstate;
 	chemstate.rho = rho;
-	chemstate.T = Tgas;
+        // Define and initialize Tgas here
+        amrex::Real Tgas_value = Tgas;
+	chemstate.T = Tgas_value;
 
 	if (massScalars.has_value()) {
 		const auto &massArray = massScalars.value();
@@ -123,7 +125,7 @@ AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE auto EOS<problem_t>::ComputeEintTempDer
 #ifdef PRIMORDIAL_CHEM
 	burn_t chemstate;
 	chemstate.rho = rho;
-	chemstate.T = Tgas;
+        // we don't need Tgas to find chemstate.dedT
 
 	if (massScalars.has_value()) {
 		const auto &massArray = massScalars.value();

--- a/src/EOS.hpp
+++ b/src/EOS.hpp
@@ -64,8 +64,12 @@ AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE auto EOS<problem_t>::ComputeTgasFromEin
 	burn_t chemstate;
 	chemstate.rho = rho;
 	chemstate.e = Eint / rho;
-	for (int nn = 0; nn < nmscalars_; ++nn) {
-		chemstate.xn[nn] = massScalars[nn] / spmasses[nn]; // massScalars are partial densities (massFractions * rho)
+
+	if (massScalars.has_value()) {
+		const auto& massArray = massScalars.value();
+		for (int nn = 0; nn < nmscalars_; ++nn) {
+			chemstate.xn[nn] = massArray[nn] / spmasses[nn]; // massScalars are partial densities (massFractions * rho)
+		}
 	}
 
 	eos(eos_input_re, chemstate);
@@ -90,8 +94,12 @@ AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE auto EOS<problem_t>::ComputeEintFromTga
 	burn_t chemstate;
 	chemstate.rho = rho;
 	chemstate.T = Tgas;
-	for (int nn = 0; nn < nmscalars_; ++nn) {
-		chemstate.xn[nn] = massScalars[nn] / spmasses[nn]; // massScalars are partial densities (massFractions * rho)
+
+	if (massScalars.has_value()) {
+		const auto& massArray = massScalars.value();
+		for (int nn = 0; nn < nmscalars_; ++nn) {
+			chemstate.xn[nn] = massArray[nn] / spmasses[nn]; // massScalars are partial densities (massFractions * rho)
+		}
 	}
 
 	eos(eos_input_rt, chemstate);
@@ -116,8 +124,12 @@ AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE auto EOS<problem_t>::ComputeEintTempDer
 	burn_t chemstate;
 	chemstate.rho = rho;
 	chemstate.T = Tgas;
-	for (int nn = 0; nn < nmscalars_; ++nn) {
-		chemstate.xn[nn] = massScalars[nn] / spmasses[nn]; // massScalars are partial densities (massFractions * rho)
+
+	if (massScalars.has_value()) {
+		const auto& massArray = massScalars.value();
+		for (int nn = 0; nn < nmscalars_; ++nn) {
+			chemstate.xn[nn] = massArray[nn] / spmasses[nn]; // massScalars are partial densities (massFractions * rho)
+		}
 	}
 
 	eos(eos_input_rt, chemstate);

